Connection: switch the "plugin activated" banner to the common connection flow

sergeymitr opened the pull request:
## Proposed changes:
- The connection banner that appears after plugin activation relies on the old connection flow, and sends API requests itself. We'll switch it to using the new "Connection Screen" flow.
- Make the Connection Screen utilize the `from` parameter, and pass it further to Calypso.
- Remove a bunch of legacy code that's no longer needed.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
1191179647901802-as-1203777512323068

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Disconnect and deactivate Jetpack.
2. Go to "Plugins", activate Jetpack. You should see the full-screen connection banner, confirm that it looks good: <img width="858" alt="Screen Shot 2023-03-16 at 21 24 17" src="https://user-images.githubusercontent.com/1341249/225788402-c5828565-290f-413b-ba38-55c7a539bd9a.png">
3. Click "Set up Jetpack", the button should turn grey, and you should get redirected to the connection screen with the flow already started.
4. Finish the connection, confirm it went smoothly.
5. Disconnect, go to the WP-admin Dashboard (not Jetpack Dashboard), you should see the top banner, confirm it looks good: <img width="1303" alt="Screen Shot 2023-03-16 at 21 28 14" src="https://user-images.githubusercontent.com/1341249/225788874-8da33a27-19bd-4c1d-b2ea-3121a8c25177.png">
6. Click "Set up Jetpack", the button should turn grey, and you should get redirected to the connection screen with the flow already started.
7. Finish the flow, confirm it goes smoothly.

sergeymitr committed:
Connection: switch the plugin activation banner to the common flow.

sergeymitr committed:
Add changelog.

sergeymitr committed:
Some clean up.

darssen commented:
Code looks good, although would like someone with more background to check all the removed legacy code.

Tested for JN site with the branch and test instructions work smoothly.

sergeymitr commented:
Hi @darssen, thanks for taking a look!
I agree, the more reviews the better, I wouldn't want to accidentally break some connection flows 🙂 

Here's a bit more context.

The banner used to make its own API request to register the site, so we removed that. Instead, we simply redirect users to `admin.php?page=jetpack#/setup`, which initiates the connection process automatically.

Some of the removed code was left from the long gone In-Place Connection (connecting users in iframe). We disabled it a while ago, and removed most of the code, but not all of it.

darssen reviewed (approved):
LGTM!
